client/src/components/CardButton.js

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components/macro";
+
+/**
+ * Primary UI component for user interaction
+ */
+const types = {
+  rider: {
+    color: "var(--text-primary)",
+    background: "transparent",
+  },
+  timer: {
+    color: "var(--text-secondary)",
+    background: "var(--text-primary)",
+  },
+  info: {
+    color: "var(--text-secondary)",
+    background: "var(--text-primary)",
+    label: "Info",
+  },
+};
+
+const ButtonElement = styled.div`
+  padding: ${(props) =>
+      props.type === "rider" || props.type === "timer" ? "0.2rem" : "0.7rem"}
+    0.8rem;
+  height: ${(props) =>
+    props.type === "rider" || props.type === "timer" ? "30px" : "40px"};
+
+  line-height: ${(props) =>
+    props.type === "rider" || props.type === "timer" ? "1.5" : "1.3"};
+
+  font-family: ${(props) => (props.type === "info" ? "Goldman" : "inherit")};
+  font-weight: ${(props) =>
+    props.type === "timer" || props.type === "rider" ? "normal" : "bold"};
+
+  color: ${(props) => types[props.type].color || types.default.color};
+  background-color: ${(props) =>
+    types[props.type].background || types.default.background};
+
+  border-radius: var(--border-radius);
+  box-shadow: ${(props) =>
+    props.type === "timer" ? "var(--insetShadow)" : "var(--shadow)"};
+`;
+
+export const CardButton = ({ type, label, onClick }) => {
+  label = types[type].label || label;
```

Comment:
@Leon you said something to this line last time ... i just cant remember what it was :)

Reply:
Yes :), it's about [immutability](https://reactjs.org/docs/optimizing-performance.html#the-power-of-not-mutating-data).

Example:

```jsx

  const label = "bwnstck"

  const App =() =>{
    useEffect(() => console.log({ type }), []);

    return <CardButton type="info" label={label} />
  };
```

What is the output of the console.log?
It is `"Info"` because you change the value of `label` inside the `CardButton`.
This is risky and in different scenario, you might update a state by accident.

Thus, you should not reassign `label`.

```suggestion
  const buttonLabel = types[type].label || label;
```

Reply:
ahh interesting :) 👍